jina/drivers/__init__.py

```diff
@@ -215,21 +215,25 @@ def __getstate__(self) -> Dict[str, Any]:
 
 class BaseRecursiveDriver(BaseDriver):
 
-    def __init__(self, recur_range: Tuple[int] = (0, 1), apply_order: str = 'post',
-                 recur_on: Tuple[str] = ('chunks',), *args, **kwargs):
+    def __init__(self,
+                 recur_depth_range: Tuple[int, int] = (0, 1),
+                 recur_adjacency_range: Tuple[int, int] = (0, 0),
```

Comment:
```suggestion
                 recur_adjacency_range: Tuple[int, int] = (0, 1),
```

Reply:
My concern with this is that it is like setting `recur_on: (chunks, matches)` by default.

Reply:
I think adding this a lot of tests will fail